rust-src/dodis_yampolskiy_prf/src/errors.rs

```diff
@@ -1,38 +1,22 @@
 // rustc seems to think the typenames in match statements (e.g. in
 // Display) should be snake cased, for some reason.
 
-use core::fmt::{self, Display};
+use thiserror::Error;
 
 /// Internal errors.  
 
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub(crate) enum InternalError {
+    #[error("Division by zero.")]
     DivisionByZero,
 }
 
-impl Display for InternalError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match *self {
-            InternalError::DivisionByZero => write!(f, "Division By Zero"),
-        }
-    }
-}
-
-impl ::failure::Fail for InternalError {}
-
 /// Errors which may occur while processing proofs and keys.
 ///
 /// This error may arise due to:
 ///
 /// * A problem with the format of `s`, a scalar,
 
-#[derive(Debug)]
+#[derive(Debug, Error)]
+#[error("{0}")]
```

Comment:
Please document what this annotation does to the Display implementation.

Also, would `error(transparent)`  not be clearer?